File: App_Clint/lib/features/reviews/domain/entities/review.dart
import 'package:equatable/equatable.dart';

class Review extends Equatable {
  const Review({
    required this.id,
    required this.bookingId,
    required this.userId,
    this.userName,
    required this.workerId,
    this.workerName,
    required this.rating,
    this.comment,
    required this.createdAt,
  });

  final int id;
  final int bookingId;
  final int userId;
  final String? userName;
  final int workerId;
  final String? workerName;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  @override
  List<Object?> get props => [
        id,
        bookingId,
        userId,
        userName,
        workerId,
        workerName,
        rating,
        comment,
        createdAt,
      ];
}

double averageReviewRating(List<Review> reviews) {
  if (reviews.isEmpty) return 0;
  final sum = reviews.fold<int>(0, (acc, r) => acc + r.rating);
  return sum / reviews.length;
}

File: Backend API/CleaningHouse_API/Models/Customers/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CleaningHouse_API.Models.Common;
using CleaningHouse_API.Models.Companies;

namespace CleaningHouse_API.Models.Customers;

[Table("Reviews")]
public class Review
{
    [Key]
    public int Id { get; set; }

    [Required]
    public int BookingId { get; set; }

    [Required]
    public int UserId { get; set; }

    [Required]
    public int WorkerId { get; set; }

    [Required]
    public int Rating { get; set; }

    [MaxLength(1000)]
    public string? Comment { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey(nameof(BookingId))]
    public virtual Booking? Booking { get; set; }

    [ForeignKey(nameof(UserId))]
    public virtual AppUser? AppUser { get; set; }

    [ForeignKey(nameof(WorkerId))]
    public virtual Worker? Worker { get; set; }
}
